crates/agent/src/main.rs

```diff
@@ -47,9 +49,13 @@ async fn main() -> agent::Result<()> {
         }
     };
 
-    let agent_service = AgentService::new(db, config.clone());
+    let agent_service = AgentService::new(db.clone(), config.clone());
```

Comment:
To avoid cloning db repeatedly
```suggestion
let agent_service = AgentService::new(Arc::clone(&db), config.clone());
```

Reply:
I've made the cache non optional